src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -87,102 +111,235 @@ export type ShowBannerWithExpiryTime = {
     expiry?: number;
 };
 @injectable()
-export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
-    public get enabled(): boolean {
-        if (this.applicationEnvironment.uiKind !== UIKind.Desktop || this.applicationEnvironment.channel !== 'stable') {
+export class DataScienceSurveyBanner implements IJupyterExtensionBanner, IExtensionSingleActivationService {
+    public isEnabled(type: BannerType): boolean {
+        switch (type) {
+            case BannerType.InsidersNotebookSurvey:
+                if (this.useVSCodeNotebookEditorApi && this.applicationEnvironment.channel === 'insiders') {
+                    return this.isEnabledInternal(type);
+                }
+                break;
+            case BannerType.ExperimentNotebookSurvey:
+                if (
+                    this.applicationEnvironment.channel === 'stable' &&
+                    this.experimentService.inExperiment(Experiments.NativeNotebook)
+                ) {
+                    return this.isEnabledInternal(type);
+                }
+                break;
+            case BannerType.DSSurvey:
+            default:
```

Comment:
Just a minor thing here. I'd break off default here and have it traceError and return false. isEnabled should always be called with a BannerType, so if you want the DSSurvey you should specifically request it. If a new BannerType is added we want someone to explicitly add a new case here, now with default it would go down the DSSurvey path if you forgot to add the case, which you don't want.

Reply:
just isEnabled? or all of the switches?

Reply:
Yeah, you don't want a default like that here. Only if it has a legit default that you want to happen.